packages/cli/src/controller/publish-controller.test.ts

```diff
@@ -0,0 +1,56 @@
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import childProcess from 'child_process';
+import {createProject} from './init-controller';
+import {uploadToIpfs} from './publish-controller';
+import Build from '../commands/build';
+import Codegen from '../commands/codegen';
+
+const projectSpec = {
+  name: 'mocked_starter',
+  repository: '',
+  endpoint: 'wss://rpc.polkadot.io/public-ws',
+  author: 'jay',
+  description: 'this is test for init controller',
+  version: '',
+  license: '',
+};
+
+jest.setTimeout(90000);
+
+describe('Cli publish', () => {
+  let projectDir: string;
+
+  beforeEach(async () => {
+    const tmpdir = await fs.promises.mkdtemp(`${os.tmpdir()}${path.sep}`);
+    projectDir = path.join(tmpdir, projectSpec.name);
+
+    await createProject(tmpdir, projectSpec);
+
+    // Install dependencies
+    childProcess.execSync('npm i', {cwd: projectDir});
+
+    await Codegen.run(['-l', projectDir]);
+    await Build.run(['-l', projectDir]);
+  });
+
+  afterEach(() => {
+    childProcess.execSync(`rm -rf ${projectDir}`);
+  });
+
+  it('should upload appropriate files to IPFS', async () => {
+    const results = await uploadToIpfs('http://localhost:5001/api/v0', projectDir);
```

Comment:
This will fail for [this](https://github.com/subquery/subql/pull/486/files#r711821190) reason